src/service_control_client_impl_quota_test.cc

```diff
@@ -273,26 +275,50 @@ TEST_F(ServiceControlClientImplQuotaTest, TestCachedQuotaRefreshGotHTTPError) {
   // the quota cache refresh.
   cached_client_->Quota(quota_request1_, &quota_response,
       [&done_status](Status status) { done_status = status; });
-  // Check the cached response is negative
+  // Check the cached response is the first negative response - QUOTA_EXHAUSTED.
   EXPECT_EQ(quota_response.allocate_errors_size(), 1);
+  EXPECT_EQ(quota_response.allocate_errors()[0].code(),
+            error_quota_response1_.allocate_errors[0].code());
 
   // AllocateQuotaFlushCallback replaces the cached negative response with
-  // the temporary positive one for failed open on transport callback error
+  // the newly cached negative response - CancelledError.
 
-  // Read the positive response from the cache
+  // Set the callback function to return a fail-open negative status, e.g.
+  // InternalError. A fail-open dummy response will be cached as a positive
+  // response.
+  mock_quota_transport_.done_status_ = absl::InternalError("internal error");
+
+  // Wait 600ms to trigger the next quota refresh
+  std::this_thread::sleep_for(std::chrono::milliseconds(600));
+
+  // Read the CancelledError response from the cache, and trigger another
+  // quota cache refresh.
   cached_client_->Quota(
       quota_request1_, &quota_response,
       [&done_status](Status status) { done_status = status; });
 
-  // Check the cached response is positive
+  // Check the returned response is the previously cached CancelledError response.
+  // A fail-close error response is cached as:
+  // AllocateQuoteResponse{allocate_errors:[QuotaError{code:UNSPECIFIED,status:error_status}]}
+  EXPECT_EQ(quota_response.allocate_errors_size(), 1);
+  EXPECT_EQ(quota_response.allocate_errors()[0].status().code(),
+            (int)cancelledError.code());
+  EXPECT_EQ(quota_response.allocate_errors()[0].status().message(),
```

Comment:
This will return previous cached "cancel" response, since it is a negative response, it will trigger a new remote quota call
the new fail-open "internal-error" response will be cached right after.